### ISSUE TITLE
non-open source submission

### ISSUE THREAD

**Issue by dvolgyes** [Sjyhne/MapAI-Competition#6]
Hi,

We are a small company, and we would like to participate, to some degree, let me explain.
So if I understand it correctly, the training and inference codes should be submitted 
with the merge request along with a binary model (pytorch, onnx, something) which
can make predictions, and this would be used to evaluate the hidden test set.

We have the following issue:
- we would be happy to submit technical description
- and trained onnx or pytorch model, 

But we have quite a large training infrastructure.
This includes hyperparameter search, training codes, etc.
This code is proprietary, and we cannot share it.

So my questions are:
- is it possible to submit a onnx or similar model which is trained but not trainable?

Our goal is not to take the prize, but to compare our capabilities to the other submissions.
If we get good enough, we would be happy to present our approach, 
but without open sourcing our training infrastructure.

In other competitions, it is frequently the case that the non-open source 
submissions are not eligible for prizes, etc., but submission is allowed, 
and shown on a leaderboard, etc.
so I would like to get some clarification on what would be the case here.

**Comment by Sjyhne** [comment]
Hello!

To be able to compete for the prizes, the source code will have to be open-source, as you suspect. However, you are more than welcome to submit a model for inference/testing to "benchmark" your model with the other submissions! I'd also appreciate a technical report of the model/approach as well :smile: 

So to clarify;
- You are not eligible for prizes
- You are allowed to submit and appear on the leaderboard
- I would appreciate the submission of a technical report on the model and training that you can share

Appreciating the interest in participation! :smiley:

**Comment by dvolgyes** [comment]
Thank you for the clear answers, these conditions look fine to us.

We are not afraid of sharing the technical details, like outlining it in a paper,
however, the source codes are literally hundreds of files in several repositories, 
so sharing them is both technically hard and it is also core IP of the company.

Anyhow, we will try to dedicate time to the competition, and make a submission which is
competitive but not eligible. 

One remark for next time: Lidar with or without aerial imagery is a bit unfortunate formulation. 
One can construct a network which takes the merged input, drops lidar, and just uses aerial imagery.
For next time I would recommend: three categories: aerial only, lidar only, overall best 
(which might use combination but also might just rely on one of the two sources)

Interesting exercise could be: taking the top N models (3-5), and merge their predictions, 
and test how the ensemble of the participants relate to the best model. When the competition is 
close race, this ensemble frequently outperforms the best model. When there is an outstanding model,
averaging with less performant ones usually reduces the output.

**Comment by Sjyhne** [comment]
That sounds good. Thank you.

About the remark - I see your point. However, I do not believe that only aerial images will be able to perform better than aerial images + lidar. Therefore, yes, they can drop the lidar input, but they will most likely perform worse than the others that use the lidar with the images. However, I note your remark and will be more specific next time.

About the exercise - Yes, absolutely interesting! I will definitely check this out.

Thanks.
